nomad/structs/structs.go

```diff
@@ -11725,9 +11726,10 @@ func NewIdentityClaims(job *Job, alloc *Allocation, wihandle *WIHandle, wid *Wor
 
 	jwtnow := jwt.NewNumericDate(now.UTC())
 	claims := &IdentityClaims{
-		Namespace:    alloc.Namespace,
-		JobID:        alloc.JobID,
-		AllocationID: alloc.ID,
+		NamespacedJobID: alloc.Namespace + "::" + alloc.JobID,
```

Comment:
Is there a functional reason for double colons (`::`)? The `sub` claim uses a single colon as a separator.

Should we prepend `alloc.Region` as well for circumstances where multiple Nomad clusters use the same Vault instance?